### DISCUSSION TITLE
Missing property

### DISCUSSION THREAD

**Discussion by bakerhaider** [swellstores/community#472] [Help] [answered]
Hello, I have this error which im not sure if its caused by Swell or TS. `const variantArray: Variant[] = prod?.variants?.results;` which gives this error: Property 'results' does not exist on type 'Variant[]'.ts(2339)
any. Im using the 'Variant' types from Swell-js. 
![image](https://github.com/swellstores/community/assets/33936834/90bc5ea5-3f4a-4652-b912-2ad24eb56176)

**Comment by logeshswell** [comment, marked as answer]
I believe `prod?.variants` return Variant[]. 
https://github.com/swellstores/swell-js/blob/c7b6f633c979d52ebe9cb757259bbba42ca7a91e/types/product/snake.d.ts#L183

You don't need to fetch `results` value additionally.

**Reply by bakerhaider** [reply]
Ok so ive removed .results and now i get this error instead: Unhandled Runtime Error
TypeError: variantArray.find is not a function

**Reply by logeshswell** [reply]
Which swell-js version do you use?

**Reply by bakerhaider** [reply]
"swell-js": "^3.22.4",

**Reply by bakerhaider** [reply]
ok so ive updated to  "^3.24.5", but still get error

**Reply by logeshswell** [reply]
I couldn't reproduce this same error with the same swell-js version. Can you try checking the type of the variable: `variantArray`?
`typeof variantArray`

**Reply by bakerhaider** [reply]
thanks for the replies! Its being logged as 'object'
